crates/sui-core/src/safe_client.rs

```diff
@@ -47,6 +49,16 @@ impl<C> SafeClient<C> {
         &mut self.authority_client
     }
 
+    fn get_committee(&self, epoch_id: &EpochId) -> SuiResult<Committee> {
+        match self.epoch_store.get_authenticated_epoch(epoch_id)? {
```

Comment:
Does that do a DB read? If so I would try to minimise its repeated use in the functions (call it once at the start).